src/circuit/gadgets/constraints.rs

```diff
@@ -1062,5 +1063,29 @@ mod tests {
             assert!(cs.is_satisfied());
         }
 
+        #[test]
+        fn prop_enforce_implication_lc((premise_val, lc_val) in any::<(bool, bool)>()) {
+            let mut cs = TestConstraintSystem::<Fr>::new();
+
+            let mut lc = LinearCombination::zero();
+            lc = lc + (if lc_val { Fr::ONE } else { Fr::ZERO }, TestConstraintSystem::<Fr>::one());
```

Comment:
On consideration, I think it would make sense to test one more case of the lc. Specifically, we should show that if premise is true and lc is some non-zero, non-one value, the constraint system is not satisfied.